docs/se/guides/migration_3x.adoc

```diff
@@ -0,0 +1,266 @@
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Helidon SE 3.x Upgrade Guide
+:description: Helidon SE 3.x Upgrade Guide
+:keywords: helidon, porting, migration, upgrade, incompatibilities
+:rootdir: {docdir}/../..
+
+In Helidon 3 we have made some changes to APIs and runtime behavior. This guide
+will help you upgrade a Helidon SE 2.x application to 3.x.
+
+
+== Java 17 Runtime
+
+Java 11 is no longer supported in Helidon 3. Java 17 or newer is required. Please follow the instructions in xref:{rootdir}/about/prerequisites.adoc[Prerequisites] for proper installation.
+
+== New Routing
+
+Handling routes based on the protocol version is now possible by registering specific routes
+on routing builder.
+
+For further information check xref:../webserver.adoc[WebServer Documentation]
+
+== Http/2 Support
+
+Helidon support of Http/2 is no longer experimental.
+
+Http/2 needed to be explicitly enabled by configuration in Helidon 2:
+
+.Enabling Http/2 support in Helidon 2
+[source,yaml]
+----
+server:
+  port: 8080
+  host: 0.0.0.0
+  experimental:
+    enable-http2: true
+    http2-max-content-length: 16384
+----
+
+In Helidon 3 Http/2 is automatically enabled when artifact with Http/2 support is
+available on the classpath.
+
+.Enabling Http/2 support in Helidon 3 by adding dependency
+[source,xml]
+----
+<dependency>
+    <groupId>io.helidon.webserver</groupId>
+    <artifactId>helidon-webserver-http2</artifactId>
+</dependency>
+----
+
+With above dependency Helidon 3 supports Http/2 upgrade from Http/1,
+cleartext Http/2 without prior knowledge and Http/2 with ALPN over TLS.
+
+In Helidon 2, max content length was configurable with `server.experimental.http2-max-content-length`, in Helidon 3 can be configured with `server.max-upgrade-content-length` globally or per socket with the same `max-upgrade-content-length` key.
+
+.Max upgrade content length in Helidon 3
+[source,yaml]
+----
+server:
+  port: 8080
+  host: 0.0.0.0
+  max-upgrade-content-length: 16384
+----
+
+For further information check xref:../webserver.adoc[WebServer Documentation]
+
+== WebSocket
+
+Helidon SE support is now based on the `WebSocketRouting` class which enables Helidon application to
+configure routing for both annotated and programmatic WebSocket endpoints.
+`TyrusSupport` is now deprecated. Websocket support in now placed in different artifact.
```

Comment:
WebSocket support is now placed in a different artifact.

Reply:
fixed